test/e2e/util_test.go

```diff
@@ -882,3 +884,104 @@ func toggleFeatureGates(deployment *appsv1.Deployment, toToggle ...featuregate.F
 
 	awaitPredicates(deadline, w, deploymentReplicas(2), deploymentAvailable, deploymentReplicas(1))
 }
+
+type Object interface {
+	runtime.Object
+	metav1.Object
+}
+
+// Apply returns a function that invokes a change func on an object and performs a server-side apply patch with the result and its status subresource.
+// The given resource must only specify its Name, Namespace, APIVersion, and Kind.
+// The given change function must be aunary, matching the signature: "func(<obj type>) error".
+// The returned function is suitable for use w/ asyncronous assertions.
+// Ex. Change the spec of an existing InstallPlan
+//
+// plan := &InstallPlan{}
+// plan.SetNamespace("ns")
+// plan.SetName("install-123def")
+// plan.GetObjectKind().SetGroupVersionKind(schema.GroupVersionKind{
+//		Group:   GroupName,
+//		Version: GroupVersion,
+//		Kind:    InstallPlanKind,
+// })
+// Eventually(Apply(plan, func(p *v1alpha1.InstallPlan) error {
+//		p.Spec.Approved = true
+//		return nil
+// })).Should(Succeed())
+func Apply(obj Object, changeFunc interface{}) func() error {
+	// Ensure given function matches expected signature
+	var (
+		change     = reflect.ValueOf(changeFunc)
+		changeType = change.Type()
+	)
+	if n := changeType.NumIn(); n != 1 {
+		panic(fmt.Sprintf("unexpected number of formal parameters in change function signature: expected 1, present %d", n))
+	}
+	if at, pt := reflect.TypeOf(obj), changeType.In(0); at != pt {
+		panic(fmt.Sprintf("argument type does not match the change function parameter type: argument %s, parameter: %s", at, pt))
+	}
+	if n := changeType.NumOut(); n != 1 {
+		panic(fmt.Sprintf("unexpected number of return values in change function signature: expected 1, present %d", n))
+	}
+	var err error
+	if rt := changeType.Out(0); !rt.Implements(reflect.TypeOf((*error)(nil)).Elem()) {
+		panic(fmt.Sprintf("unexpected return type in change function signature: expected %t, preset %s", err, rt))
+	}
+
+	var (
+		bg     = context.Background()
+		client = ctx.Ctx().Client()
+	)
+	key, err := controllerclient.ObjectKeyFromObject(obj)
+	if err != nil {
+		panic(fmt.Sprintf("unable to extract key from resource: %s", err))
+	}
+
+	return func() error {
+		cp := obj.DeepCopyObject().(Object)
+		if err := client.Get(bg, key, cp); err != nil {
+			return err
+		}
+		cp.GetObjectKind().SetGroupVersionKind(obj.GetObjectKind().GroupVersionKind())
+		cp.SetManagedFields(nil)
```

Comment:
is this needed? shouldn't the `ForceOwnership` make `test` become the owner of any changed fields?

Reply:
this came out of some red/green/refactor-ing -- an error is returned without it.